http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -507,43 +501,15 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
   }
 
   override def deleteAndPollRuntime(msg: DeleteAzureRuntimeMessage)(implicit ev: Ask[F, AppContext]): F[Unit] = {
-    implicit val azureRuntimeDeletingDoneCheckable: DoneCheckable[Option[VirtualMachine]] =
-      (v: Option[VirtualMachine]) => v.isEmpty
+    implicit val wsmDeleteVmDoneCheckable: DoneCheckable[GetDeleteJobResult] = (v: GetDeleteJobResult) =>
+      v.jobReport.status.equals(WsmJobStatus.Succeeded) || v.jobReport.status == WsmJobStatus.Failed
     for {
       ctx <- ev.ask
 
       runtimeOpt <- dbRef.inTransaction(clusterQuery.getClusterById(msg.runtimeId))
       runtime <- F.fromOption(runtimeOpt, PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
       auth <- samDAO.getLeoAuthToken
 
-      // Delete the VM in WSM
```

Comment:
here I basically moved the WSM job polling into the wsmResourceId fold because I added an adaptError to the deleteDisk call which, in the case that the wsmResourceId was None and the deleteDisk call errored, the runtime would be updated to 'Deleted' and the destroyedDate updated, but then the error would be caught by the handleAzureRuntimeDeletionError, which then set the status to 'Error'.